### ISSUE TITLE
Implement text-emphasis handling

### ISSUE THREAD

**Issue by Gankro** [servo/servo#17715]
https://developer.mozilla.org/en-US/docs/Web/CSS/text-emphasis

* stylo already handles it (because gecko does)
* webrender doesn't need anything to handle it

The layout code just needs to grab the info from stylo, figure out the positioning of the glyphs, and emit a TextDisplayItem into its DisplayList.

I have marked off where text-emphasis marks should (potentially) be emitted in this PR: https://github.com/servo/servo/pull/17675.

**Comment by Gankro** [comment]
I guess one subtlety is that text-emphasis marks require space so you may need to adjust the whole text's baseline(?) to make room.

**Comment by askalski** [comment]
Don't assign me yet, but I plan to give it a look.

**Comment by NeverHappened** [comment]
Hi! Can you assign me? I'm not a contributor yet though

**Comment by KiChjang** [comment]
@askalski Have you begun working on this issue? Would it be OK for @NeverHappened to start working on it?

**Comment by NeverHappened** [comment]
I've done some work on this issue and have questions. Where can I ask them? In IRC? Who should I talk to?

**Comment by jdm** [comment]
IRC or here are fine choices, especially if you don't end up receiving a response on IRC.

**Comment by NeverHappened** [comment]
1. Why do we need this feature? As I understand, only Firefox has it :)
2. As I understand, the text emphasis should be at the top and be 50% of the size of the base text. But I can't find any code that is able to scale/clone the TextDisplayItem, or easily generate new one? Does this require a lot of new code?

**Comment by jdm** [comment]
1. It's a CSS standard, and supported by Safari according to http://caniuse.com/#feat=text-emphasis .
2. @mrobinson is this a question you can answer?

**Comment by mrobinson** [comment]
@NeverHappened Any reason you need to clone the `TextDisplayItem` rather than just creating a new one with the properties calculated for the emphasis?

**Comment by NeverHappened** [comment]
No, no reason. I have issues because the code for generating display item is in the building display list phase, but the calculating text properties is done long before that.

**Comment by hieudo-dev** [comment]
Hi all! Is there anyone working on this ? I'd love to give this one a try if no one else is
